service/dap/server.go

```diff
@@ -357,18 +357,20 @@ func (s *Server) serveDAPCodec() {
 	s.reader = bufio.NewReader(s.conn)
 	for {
 		request, err := dap.ReadProtocolMessage(s.reader)
-		// TODO(polina): Differentiate between errors and handle them
```

Comment:
Clarification about graceful error handling. There were 2 original todos that now got removed. The 2nd one was specifically for what you are adding, so makes sense. I am wondering about the 1st one. It was referring to the fact that there might be other kinds of errors here that we might want to handle gracefully. "Unknown command" was just one of them.  Technically, the way your code is written right now it would cover more than just unknown request commands , so "for example" that got removed is still accurate. Those other ones are just not at all likely (like unknown type). But in addition to the DecodeProtocolMessageFieldError, there could other kind of decoding errors. Maybe also network errors? Makes sense to make those fatal, but what bout thing like requests that are too large or unmarshalling errors? Would we want to handle those gracefully? Not that you need to add any code now, jus wondering if it makes sense to keep and revisit the TODO.

Reply:
Good point. I will add more details back into this comment.